modules/default/weather/providers/weathergov.js

```diff
@@ -0,0 +1,270 @@
+	// This isn't strictly necessary, since it will fallback to the provider identifier
+	// But for debugging (and future alerts) it would be nice to have the real name.
+	providerName: "Weather.gov",
+
+	// Overwrite the fetchCurrentWeather method.
+	fetchCurrentWeather() {
+		this.fetchData(this.getUrl())
+			.then(data => {
+				if (!data || !data.properties || !data.properties.periods || !data.properties.periods.length) {
+					// Did not receive usable new data.
+					// Maybe this needs a better check?
+					return;
+				}
+
+				//this.setFetchedLocation(`${data.name}, ${data.sys.country}`);
+
+				const currentWeather = this.generateWeatherObjectFromCurrentWeather(data.properties.periods[0]);
+				this.setCurrentWeather(currentWeather);
+			})
+			.catch(function(request) {
+				Log.error("Could not load data ... ", request);
+			})
+	},
+
+	// Overwrite the fetchCurrentWeather method.
+	fetchWeatherForecast() {
+		this.fetchData(this.getUrl())
+			.then(data => {
+				if (!data || !data.properties || !data.properties.periods || !data.properties.periods.length) {
+					// Did not receive usable new data.
+					// Maybe this needs a better check?
+					return;
+				}
+
+				//this.setFetchedLocation(`${data.city.name}, ${data.city.country}`);
+
+				const forecast = this.generateWeatherObjectsFromForecast(data.properties.periods);
+				this.setWeatherForecast(forecast);
+			})
+			.catch(function(request) {
+				Log.error("Could not load data ... ", request);
+			})
+	},
+
+	/** Weather.gov Specific Methods - These are not part of the default provider methods */
+	/*
+	 * Gets the complete url for the request
+	 */
+	getUrl() {
+		return this.config.apiBase + this.config.lat + "," + this.config.lon + this.config.weatherEndpoint;
+	},
+
+	/*
+	 * Generate a WeatherObject based on currentWeatherInformation
+	 */
+	generateWeatherObjectFromCurrentWeather(currentWeatherData) {
+		const currentWeather = new WeatherObject(this.config.units);
+
+		//currentWeather.humidity = currentWeatherData.main.humidity;
+		currentWeather.temperature = currentWeatherData.temperature;
+		currentWeather.windSpeed = currentWeatherData.windSpeed.split(" ", 1);
+		currentWeather.windDirection = this.convertDirectiontoDegrees(currentWeatherData.windDirection);
+		currentWeather.weatherType = this.convertWeatherType(currentWeatherData.shortForecast, currentWeatherData.isDaytime);
+		//currentWeather.sunrise = moment(currentWeatherData.sys.sunrise, "X");
+		//currentWeather.sunset = moment(currentWeatherData.sys.sunset, "X");
+
+		return currentWeather;
+	},
+
+	/*
+	 * Generate WeatherObjects based on forecast information
+	 */
+	generateWeatherObjectsFromForecast(forecasts) {
+		return this.fetchForecastDaily(forecasts);
+	},
+
+	/*
+	 * fetch forecast information for daily forecast.
+	 */
+	fetchForecastDaily(forecasts) {
+		// initial variable declaration
+		const days = [];
+		// variables for temperature range and rain
+		var minTemp = [];
```

Comment:
please do not mix `var` with `const` and `let` in this function. Stick to `const` and `let` like in the rest of the file.